File: spark-project/web/src/main/java/com/aura/util/StringUtil.java
package com.aura.util;

/**
 * @项目名称：专家系统
 * @类名称：StringUtil
 * @类描述：字符工具类
 * @创建人：
 * @创建时间：2012-09-07
 * @修改人：
 * @修改时间：
 * @备注：
 */
public class StringUtil {
	
	/**
	 * 字符串首字母大写
	 * @param 字符串
	 * @return 首字母大写的字符串
	 */
	public static String toFirstUpperCase(String str) {
		if(str == null || str.length() < 1) {
			return "";
		}
		String start = str.substring(0,1).toUpperCase();
		String end = str.substring(1, str.length());
		return start + end;
	}
	/**
	 * 标题长度截取
	 * @param title
	 * @return
	 */
	public static String limitTitle(String title,int count) {
		if(title != null && title.length() > count) {
			return title.substring(0, count) + "...";
		}
		return title;
	}
	/**
	 * 标题清理
	 * @param title
	 * @return
	 */
	public static String clearTitleAll(String title) {
		String result = clearTitle(title, "-");
		result = clearTitle(result, "_");
		return result;
	}
	/**
	 * 标题清理
	 * @param title
	 * @param symbol
	 * @return
	 */
	public static String clearTitle(String title, String symbol) {
		if(title == null) {
			return "";
		}
		int index = title.lastIndexOf(symbol);
		return index < 5 ? title : clearTitle(title.substring(0, index), symbol);
	}
	
	public static void main(String [] args) {
		String title = "习近平访问欧盟";
		System.out.println(StringUtil.limitTitle(title, 5));
	}
}

File: spark-project/analysis/src/main/scala/com/aura/util/StringUtil.scala
package com.aura.util

object StringUtil {

	/**
		* 标题和Url截断
		* @param str
		* @param len
		* @param encoding
		* @return
		*/
	def limitString(str: String , len: Int , encoding: String): String = {
		  var s = str
			var yy: Array[Byte] = s.getBytes(encoding)
			while(s != null && yy.length > len) {
				var sublen: Int  = getEncodingLen(yy.length, s.length(), len)
				if(sublen < 0) {
					sublen = s.length() / 2
				}
				s = s.substring(0, sublen)
				yy = s.getBytes(encoding)
			}
			s
	}
	
	def getEncodingLen(bytelen: Int, strlen: Int , maxlen: Int): Int = {
		strlen - (bytelen - maxlen)
	}

	/**
		* 标题清理过滤
		* @param title
		* @return
		*/
	def clearTitleAll(title: String): String = {
		var result: String = clearTitle(title, "-")
		result = clearTitle(result, "_")
		result
	}

	def clearTitle(title: String, symbol: String): String = {
		if(title == null) {
			return ""
		}
		val index: Int = title.lastIndexOf(symbol)
		if(index < 5) {
			title
		} else {
			clearTitle(title.substring(0, index), symbol)
		}
	}
}